backend/src/main/java/com/rota/api/StaffController.java

```diff
@@ -23,6 +24,7 @@
 import org.springframework.web.server.ResponseStatusException;
 
 @RestController
+@CrossOrigin(value = "*")
```

Comment:
I couldn't find which branch this was on, so wanted to flag it just incase. Something like this disables CORS, right? It strikes me as a bad idea to put this at the top of a controller class, as we'd need to do so on all controller classes we create, right?

I've seen cors disabled for a list of permitted URLs setup in a securityConfig file, similar to this one on Core Competition Engine on the link below. However, there are always multiple ways to do this, as magic did something different (but I cannot understand what....). Can you have a quick google and see if any solutions jump out?


https://github.com/ScottLogic/core-competition-engine/blob/e6d868dc64fbff06bbac9e153b9d6645759137b3/src/main/java/com/scottlogic/competitionEngine/authorization/TestSecurityConfig.java

Reply:
In a previous project I set up a CORS config file which allowed the front end API address only but in all rest controllers.